Vagrantfile

```diff
@@ -0,0 +1,38 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/precise64"
+  config.vm.provider "hyperv"
+  config.vm.provision :shell, privileged: false, inline: <<-SHELL
+	sudo apt-get update
+
+	# Install pyenv prerequisites
+	sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+	libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+	xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+
+	# Install pyenv
+	git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+	echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+	echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+	echo 'eval "$(pyenv init -)"' >> ~/.profile
+	export PYENV_ROOT="$HOME/.pyenv"
+	export PATH="$PYENV_ROOT/bin:$PATH"
+	eval "$(pyenv init -)"
+	pyenv install 3.6.6
+	pyenv global 3.6.6
+
+	# Install poetry
+	curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+	echo 'export PATH="$HOME/.poetry/bin:$PATH"' >> ~/.profile
+	export PATH="$HOME/.poetry/bin:$PATH"
+  SHELL
+  
+  config.trigger.after :up do |trigger|
+    trigger.name = "Launching App"
+    trigger.info = "Running the TODO application setup script"
+    trigger.run_remote = {privileged: false, inline: "
+      cd /vagrant
+      poetry install
+      poetry run flask run --host=0.0.0.0
```

Comment:
You may have noticed this confuses Vagrant's VM management because the "up" process doesn't finish so it doesn't know the state of it and you can't `vagrant halt` for example.

You could fix this by running the app as a background process, either via the gunicorn stretch goal or by wrapping this in: `nohup poetry run flask run --host=0.0.0.0 > logs.txt 2>&1 &` (and redirect output to a text file)